### ISSUE TITLE
HTTP2: Two remote server tests occasionally timing out in CI

### ISSUE THREAD

**Issue by geoffkizer** [dotnet/corefx#39545]
These are tests I'm modifying to run with HTTP2 in https://github.com/dotnet/corefx/pull/39343 but disabling for now.

Sometimes in CI, these tests timeout. Likely they are just running really slowly in CI and increasing the timeout would make them more reliable. We should investigate and confirm and change the timeout as appropriate.

PostAsync_RedirectWith307_LargePayload
PostRewindableContentUsingAuth_NoPreAuthenticate_Success

**Comment by eiriktsarpalis** [comment]
Would you be able to point to examples of failing builds? I'm going through the outerloop build history but haven't been able to find failing instances (presumably because they have been skipped since the beginning of time?). [Example 1](https://dev.azure.com/dnceng/public/_build/results?buildId=230241&view=ms.vss-test-web.build-test-results-tab&runId=5703398&resultId=267809&paneView=history) or [Example 2](https://dev.azure.com/dnceng/public/_build/results?buildId=270478&view=ms.vss-test-web.build-test-results-tab&runId=7430638&resultId=134301&paneView=history).

**Comment by eiriktsarpalis** [comment]
I think it's probably this failing pr build: https://dev.azure.com/dnceng/public/_build/results?buildId=265300&view=ms.vss-test-web.build-test-results-tab&runId=7206096&resultId=100021

**Comment by geoffkizer** [comment]
Yeah, that's one of the runs I saw. There was at least one other run on that PR (before I disabled the tests) that failed both as well.

**Comment by eiriktsarpalis** [comment]
FWIW that failure is targeting the platform handler.

**Comment by geoffkizer** [comment]
> FWIW that failure is targeting the platform handler.

Interesting. That would seem to point to the issue being either (a) CI slowness or (b) remote server slowness, as opposed to a product issue, since it seems to happen with different handlers.

**Comment by stephentoub** [comment]
I put it in a loop.  In < 100 iterations for each of a PlatformHandler, SocketsHttpHandler 1.1, and SocketsHttpHandler 2.0 test, I got hangs, e.g. taking greater than 10 minutes.  They're all stuck trying to write to the network, which suggests that the connection is full because the server has simply stopped reading.  @davidsh, where's the code for the endpoint these talk to?

**Comment by stephentoub** [comment]
I don't think this blocks 3.0.  Pushing out.

**Comment by eiriktsarpalis** [comment]
What's the best way to address this issue going forward? The test failures are long gone from the build history so there's not much context. @geoffkizer should we maybe try re-enabling tests for HTTP2 and observe how they behave?

**Comment by karelz** [comment]
@eiriktsarpalis are you working on it? If not, please unassign yourself. Thanks!

**Comment by eiriktsarpalis** [comment]
Was working on this before my trip. Can confirm that the test is timing out consistently with the platform handler on linux CI, but it does not reproduce on my linux machine.
